log/logtest/config.go

```diff
@@ -0,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logtest // import "go.opentelemetry.io/otel/log/logtest"
+
+import (
+	"go.opentelemetry.io/otel/log"
+)
+
+type config struct {
+	minSeverity log.Severity
+}
+
+func newConfig(options []Option) config {
+	var c config
+	for _, opt := range options {
+		c = opt.apply(c)
+	}
+
+	return c
+}
+
+// Option configures a [Hook].
+type Option interface {
+	apply(config) config
+}
+
+type optFunc func(config) config
+
+func (f optFunc) apply(c config) config { return f(c) }
+
+// WithMinSeverity returns an [Option] that configures the minimum severity the
+// recorder will return true for when Enabled is called.
+//
+// By default, the recorder will be enabled for all levels.
+func WithMinSeverity(l log.Severity) Option {
+	return optFunc(func(c config) config {
+		c.minSeverity = l
+		return c
+	})
+}
```

Comment:
I feel like passing min severity actually makes it more flexible, since it allows enabling some log and not others?

Reply:
I mean the user should be able to control what `recorder.Enabled` will invoke so that the user can simply:

```go
recorder := logtest.NewRecorder(logtest.WithEnabledFunc(func ctx context.Context, record log.Record) bool {
	return record.Severity() >= log.SeverityInfo
}))
```

Reply:
And I would remove `ContextWithEnabledRecorder` and `WithMinSeverity`.

Reply:
Done.